### ISSUE TITLE
Add kittingBox class abstraction and create logic around it

### ISSUE THREAD

**Issue by Aleynikovich** [Aleynikovich/iiwaTOFAS#72]
In ProgramSubroutines.java we have the method

public boolean placeAxisBox(Frame kittingBase, int workpieceId, int positionId)
    {
        Ixtur.detach();
        Gripper.attachTo(robot.getFlange());

        // Get the base frame (parent)
        ObjectFrame refBase = application.getApplicationData().getFrame("/basekitting");
        //Get the child frames (relative to parent)
        ObjectFrame taughtP1 = application.getApplicationData().getFrame("/basekitting/PlaceAxis1_1");
        ObjectFrame taughtP2 = application.getApplicationData().getFrame("/basekitting/PlaceAxis1_2");
        Logger.getInstance().debug("ROBOT_EXEC", "Taught P1: " + taughtP1.toString());
        Logger.getInstance().debug("ROBOT_EXEC", "Taught P2: " + taughtP2.toString());
        Logger.getInstance().debug("ROBOT_EXEC", "Basekitting: " + refBase.toString());
        //Create a new base frame
        Frame newBase = refBase.copyWithRedundancy();
        Logger.getInstance().debug("ROBOT_EXEC", "New Base copied from ref: " + newBase.toString());
        //Set the new base frame values with the data from the camera
        newBase.setX(kittingBase.getX());
        newBase.setY(kittingBase.getY());
        newBase.setZ(kittingBase.getZ());
        newBase.setAlphaRad(kittingBase.getAlphaRad());
        newBase.setBetaRad(kittingBase.getBetaRad());
        newBase.setGammaRad(kittingBase.getGammaRad());
        Logger.getInstance().debug("ROBOT_EXEC", "New Base after setter: " + newBase);
        //Create relative frames
        Frame relativeP1 = taughtP1.copyWithRedundancy();
        Frame relativeP2 = taughtP2.copyWithRedundancy();
        Logger.getInstance().debug("ROBOT_EXEC", "New Base: " + newBase);
        Logger.getInstance().debug("ROBOT_EXEC", "Relative P1: " + relativeP1);
        Logger.getInstance().debug("ROBOT_EXEC", "Relative P2: " + relativeP2);
        //Make the new frames children of the new base
        relativeP1.setParent(newBase);
        relativeP2.setParent(newBase);
        Logger.getInstance().debug("ROBOT_EXEC", "Relative P1 after parent: " + relativeP1);
        //Move the gripper to the relative frames with the new base
        Gripper.move(lin(relativeP1).setJointVelocityRel(0.5));
        Gripper.move(lin(relativeP2).setJointVelocityRel(0.1));

        toolController.openTool(toolController.getCurrentToolId());
        robot.move(linRel(0, 0, 500).setJointVelocityRel(0.5).setBlendingCart(50));
        robot.move(linRel(-200, -200, 200).setJointVelocityRel(0.5));

        return true;
    }

I want to create an abstracion of this method (it is currently working but im not too happy with how long it is and its not really reusable for other workpieces like drum and disk. In the robot I have a baseframe which is called basekitting. There are a few taught points relative to it, you can find them in the RoboticsAPI.data.xml file.

These frames are taught relative to the basekitting base, which will be moving. With a 3d camera, we will get a new basekitting base frame (kttingBase in the method) and then change these points relative to the camera data.

These are used for a kitting process of a box. There are 3 types of boxes. Each box can hold different kinds of workpieces. Currently we have 3 workpieces. An Axis, a Drum and a Disk.

The 3 boxes are slightly different so for each of these we will need a different trajectory and place position. Each box can hold 4 items.
two positions are for either drums/disks and other two are for axis. We will need to keep track of which position is occupied and the robot will have a method which will place the currently gripped item to that box. We need to update the parsing methods and subroutines to reflect that. Currently we are getting the programID, camera frame and the workpieceID in private static BaseCoordinateData parseBaseCoordinateData(String[] parts).

Currently we are going to work only with one box type for testing purposes so lets keep it simple, but scalable for the future.

The workflow wil be as follows. We pick any of the objects then we go to the kitting spot. The camera takes a picture of the box and then the driver will send us a request with id 123 (call to program 23) with the following parameters: frame (for the kittingBase) and workpieceID (what workpiece we currently hold and need to put into the box).

The program with id 23s job will be to check what position is free for that workpiece and put the workpiece there. Whenever we restart CommandExecutor, we will free the box (we will physically remove all the workpieces from the box) and start from 0, but while the program is running in its main loop, the robot will need to keep track of which positions have been filled to avoid collisions. If for example, we already have placed 2 axis in the box, and we have been called to place another axis into the box, the robot should refuse as there are no free places to put that axis in.

The positions to place the workpieces are under the /basekitting frame e.g. /basekitting/PlaceAxis1_1 (axis1, position1) PlaceAxis1_2 (axis1_position2, which is the drop position), PlaceAxis2_1(axis2, position1) and so on.

Feel free to separate methods and classes into different java files to keep them short and simple.

The disks/drums are placed with the GimaticCircMagnet tool. The axis will be picked with the GimaticGripperV tool.
